src/components/Calculator.jsx

```diff
@@ -1,4 +1,45 @@
+import React, { useState } from 'react';
+import calculate from './logic/calculate';
 import Buttons from './Buttons';
 
-const Calculator = () => <Buttons />;
+const Calculator = () => {
+  const [calcData, setCalcData] = useState({
+    total: null,
+    text: null,
+    operation: null,
+  });
+
+  const handleClick = (buttonName) => {
+    const newData = calculate(calcData, buttonName);
+    setCalcData(newData);
+  };
+
+  const displayResult = calcData.text || calcData.total || '0';
+
+  return (
+    <div className="wrapper">
+      <div className="screen">{displayResult}</div>
+      <Buttons className="btn light-gray" text="AC" onClick={() => handleClick('AC')} />
+      <Buttons className="btn light-gray" text="+/-" onClick={() => handleClick('+/-')} />
+      <Buttons className="btn light-gray" text="%" onClick={() => handleClick('%')} />
```

Comment:
- [ ] The modulus button is giving a strange result. For instance, 5 % 2 should be = 1 but it's giving 0.05. Kindly fix this so this button's function returns the correct result. 👍🏽